### PR TITLE
Adding username/password parameters to Socks5 config URL

### PR DESCRIPTION
**Pull request by amircybersec**
This change adds username and password params for authentication to socks5 config URL used to set up the dialer. The URL now accepts `UserInfo` parameter in form of `username:password`. For example:

`socks5://username:password@domain.com:port`

This change was tested with a socks5 proxy server by running connectivity test:

```
go run . -transport="socks5://foo:bar@pod3.secretexampledomain.com:8080" -proto tcp -resolver 8.8.8.8

{"resolver":"8.8.8.8:53","proto":"tcp","transport":"socks5://REDACTED@pod3.secretexampledomain.com:8080","time":"2024-03-23T03:34:31Z","duration_ms":42,"error":null}
```

### CODE REVIEW

**Comment by amircybersec** [comment]
I had to change `go.mod` for `x` packages to point to latest commit on the main branch `github.com/Jigsaw-Code/outline-sdk v0.0.0-20240322232535-212ef9692e98` to get the Github build actions pass. I guess we need to update this after bumping up the SDK version. Perhaps we can keep it like this until the SDK version is bumped?

File: x/go.mod [hunk 1]
```diff
@@ -3,13 +3,13 @@ module github.com/Jigsaw-Code/outline-sdk/x
 go 1.20
 
 require (
-	github.com/Jigsaw-Code/outline-sdk v0.0.14
+	github.com/Jigsaw-Code/outline-sdk v0.0.0-20240322232535-212ef9692e98
```
Comment (fortuna) [inline x/go.mod:6]: Please use the new v0.0.15

**Review by fortuna — COMMENTED** [review]
Looks good. Let's just update the dependency

File: x/go.mod (hunk 1 above)
Comment (amircybersec) [reply x/go.mod]: done!

**Review by fortuna — APPROVED** [review]

File: x/config/socks5.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package config
+
+import (
+	"net/url"
+
+	"github.com/Jigsaw-Code/outline-sdk/transport"
+	"github.com/Jigsaw-Code/outline-sdk/transport/socks5"
+)
+
+func newSOCKS5StreamDialerFromURL(innerDialer transport.StreamDialer, configURL *url.URL) (transport.StreamDialer, error) {
```
Comment (fortuna) [inline x/config/socks5.go:10]: Please update the Go doc with the new parameter. It doesn't show the credentials currently: https://pkg.go.dev/github.com/Jigsaw-Code/outline-sdk/x/config

Comment (amircybersec) [reply x/config/socks5.go:10]: @fortuna good catch! it slipped my mind. I updated the doc to include authentication params (`UserInfo`) in the URL.

**Review by fortuna — APPROVED** [review]

**Pull request merged**